tools/inkscape_extension/visicut_export.py

```diff
@@ -326,6 +330,17 @@ def get_original_filename(filename):
         VISICUTBIN = which("VisiCut.Linux", [VISICUTDIR, "/usr/share/visicut"])
 INKSCAPEBIN = which("inkscape", [INKSCAPEDIR])
 
+## Test if this inkscape is in an appimage.
+# We detect this by checking for an AppRun file, in one of the parent folders of our INKSCAPEBIN.
+# If so, replace INKSCAPEBIN with AppRun, as this is the only safe way to call inkscape.
+# (a direct call mixes libraries from the host system with the appimage, may or may not work.)
+dir = os.path.dirname(INKSCAPEBIN)
+while dir != '/':
```

Comment:
On Windows this gives an infinite loop because `dir` ends up being something else, e.g., `C:\`

Reply:
Indeed. In fact, there are better (and thus safer!) APIs available nowadays. Python's pathlib is well prepared for such a scenario:

```
from pathlib import Path

dir = Path(INKSCAPEBIN).parent
assert dir.is_dir()

# one could also use other metrics as an abort condition, but this will even catch the FUSE mountpoint
while not dir.is_mount():
    apprun = dir / "AppRun"
    if apprun.is_file() and apprun.is_executable():
        INKSCAPEBIN = apprun_path
        break
    dir = dir.parent

# it would be even easier with the parents property (an iterator)
# example:
# >>> list(Path("/ab/cd/ef/bin/visicut").parents)
# [PosixPath('/ab/cd/ef/bin'), PosixPath('/ab/cd/ef'), PosixPath('/ab/cd'), PosixPath('/ab'), PosixPath('/')
for parent in Path(INSCAPEBIN).parents:
    apprun = parent / "AppRun"
    if apprun.is_file() and apprun.is_executable():
        INKSCAPEBIN = apprun_path
        break

Reply:
Indeed, did not think about windows. Thanks for catching this!
I like the very readable .parents iterator!

Reply:
Indeed, it is very handy. I was about to write my own but then remembered to have a look into pathlib first.